src/internal.c

```diff
@@ -14195,9 +14201,6 @@ static int DoHandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
     /* Also, skip hashing the client_hello message here for DTLS. It will be
      * hashed later if the DTLS cookie is correct. */
     if (type != hello_request
-    #ifdef WOLFSSL_DTLS
-            && !(IsDtlsNotSctpMode(ssl) && type == client_hello)
-    #endif
```

Comment:
Won't this make the object save state when the client is not verified yet?

Reply:
Only temporary, if the client is truly unverified, we will send an HelloVerifyRequest and the state will be reset (the same way dtls_handshake_number, got_client_hello, etc. are reset).